contracts/external/cw-payroll-factory/src/contract.rs

```diff
@@ -0,0 +1,325 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    from_binary, to_binary, Binary, CosmosMsg, Deps, DepsMut, Env, MessageInfo, Order, Reply,
+    Response, StdResult, SubMsg, WasmMsg,
+};
+
+use cw2::set_contract_version;
+use cw20::Cw20ExecuteMsg;
+use cw20::Cw20ReceiveMsg;
+use cw_denom::CheckedDenom;
+use cw_storage_plus::Bound;
+use cw_utils::{nonpayable, parse_reply_instantiate_data};
+use cw_vesting::{
+    msg::{
+        InstantiateMsg as PayrollInstantiateMsg, QueryMsg as PayrollQueryMsg,
+        ReceiveMsg as PayrollReceiveMsg,
+    },
+    state::VestingPayment,
+};
+
+use crate::error::ContractError;
+use crate::msg::{ExecuteMsg, InstantiateMsg, MigrateMsg, QueryMsg, ReceiveMsg};
+use crate::state::{vesting_contracts, VestingContract, TMP_INSTANTIATOR_INFO};
+
+pub(crate) const CONTRACT_NAME: &str = "crates.io:cw-payroll-factory";
+pub(crate) const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+pub const INSTANTIATE_CONTRACT_REPLY_ID: u64 = 0;
+pub const DEFAULT_LIMIT: u32 = 10;
+pub const MAX_LIMIT: u32 = 50;
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    cw_ownable::initialize_owner(deps.storage, deps.api, msg.owner.as_deref())?;
+
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+    Ok(Response::new()
+        .add_attribute("method", "instantiate")
+        .add_attribute("creator", info.sender))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::Receive(msg) => execute_receive_cw20(env, deps, info, msg),
+        ExecuteMsg::InstantiatePayrollContract {
```

Comment:
nice! I like instantiating and funding through the factory. nit: call this `InstantiateNativePayrollContract`?